File: Programmers/체육복.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
	int answer = 0;
	vector<int> v(n + 2, 1);
	for (int i = 0; i < lost.size(); i++) v[lost[i]]--;
	for (int i = 0; i < reserve.size(); i++) v[reserve[i]]++;
	for (int i = 1; i <= n; i++) {
		if (v[i] == 0 && v[i - 1] == 2) {
			v[i]++;
			v[i - 1]--;
		}
		else if (v[i] == 0 && v[i + 1] == 2) {
			v[i]++;
			v[i + 1]--;
		}
	}

	for (int i = 1; i <= n; i++) {
		if (v[i] != 0) answer++;
	}

	return answer;
}

File: Programmers/python/체육복.py
def solution(n, lost, reserve):
    clothes = [1 for i in range(n + 1)]
    for i in lost:
        clothes[i] -= 1

    for i in reserve:
        clothes[i] += 1

    for i in range(1, n):
        if clothes[i] == 0 and clothes[i - 1] == 2:
            clothes[i] += 1
            clothes[i - 1] -= 1
        elif clothes[i] == 0 and clothes[i + 1] == 2:
            clothes[i] += 1
            clothes[i + 1] -= 1

    answer = 0
    for i in range(1, n + 1):
        if clothes[i] > 0:
            answer += 1

    return answer
